src/melee.cpp

```diff
@@ -533,31 +503,10 @@ inline double limit_probability( double unbounded_probability ) {
 
 double player::crit_chance( int roll_hit, int target_dodge, const item &weap ) const
 {
-    // TODO: see player.cpp ther eis the same listing of those skill!
-    int unarmed_skill = get_skill_level( skill_unarmed );
-    int bashing_skill = get_skill_level( skill_bashing );
-    int cutting_skill = get_skill_level( skill_cutting );
-    int stabbing_skill = get_skill_level( skill_stabbing );
-    int melee_skill = get_skill_level( skill_melee );
-
-    if( has_active_bionic("bio_cqb") ) {
-        unarmed_skill = 5;
-        bashing_skill = 5;
-        cutting_skill = 5;
-        stabbing_skill = 5;
-        melee_skill = 5;
-    }
     // Weapon to-hit roll
     double weapon_crit_chance = 0.5;
-    if( weap.has_flag("UNARMED_WEAPON") ) {
-        // Unarmed attack: 1/2 of unarmed skill is to-hit
-        ///\EFFECT_UNARMED increases crit chance with UNARMED_WEAPON
-        weapon_crit_chance = 0.5 + 0.05 * unarmed_skill;
-    }
-
     if( weap.type->m_to_hit > 0 ) {
-        weapon_crit_chance = std::max(
-            weapon_crit_chance, 0.5 + 0.1 * weap.type->m_to_hit );
+        weapon_crit_chance = std::max( weapon_crit_chance, 0.5 + 0.1 * weap.type->m_to_hit );
```

Comment:
Old code allowed `max( weap.type->m_to_hit, unarmed / 2 )`.
It could be a good idea to allow pure unarmed (`!is_armed()` as opposed to `unarmed_strike()`, preferably with a comment explaining it) to benefit from the max, but still require regular unarmed to only use weapon type to hit.